src/latexify/codegen/function_codegen.py

```diff
@@ -162,21 +162,49 @@ def visit_Match(self, node: ast.Match) -> str:
             if i < len(node.cases) - 1:
                 body_latex = self.visit(case.body[0])
                 cond_latex = self.visit(case.pattern)
-                case_latexes.append(
-                    body_latex + r", & \mathrm{if} \ " + subject_latex + cond_latex
-                )
+                if case.guard is not None:
+                    cond_latex = self._expression_codegen.visit(case.guard)
+                    subject_latex = ""  # getting variable from cond_latex
+
+                case_latexes.append(body_latex + r", & \mathrm{if} \ " + cond_latex)
             else:
                 case_latexes.append(
                     self.visit(node.cases[-1].body[0]) + r", & \mathrm{otherwise}"
                 )
 
-        return (
+        latex = (
             r"\left\{ \begin{array}{ll} "
             + r" \\ ".join(case_latexes)
             + r" \end{array} \right."
         )
 
+        latex_final = latex.replace("subject_name", subject_latex)
```

Comment:
Yes that is something we considered as well and would love to hear suggestions.

Since each of the visit_Match... functions cannot access the subject_latex variable(each of those functions can only have the ast node and pattern as the input), we had to find a workaround.  We tried to use python format and % to insert the subject_latex but the use of % and {} in latex made this very difficult. Do you have any suggestions for how to better implement this (or possible a better name than "subject_name" to replace)

Reply:
Ah yeah I see the dilemma, bit perplexing lol. Not entirely sure how to solve but I'll give it some thought. Thanks for tackling `match` statements btw! Really cool feature :)

Reply:
Not sure what @odashi thinks of this solution, but you could create a stack of `_subject_latexes: listr[str] = []`, and at the very beginning of `visit_Match` do something like

```python
subject_latex = self._expression_codegen.visit(node.subject)
self._subject_latexes.append(subject_latex)
```

and then right before you return do something like

```python
latex = (
    r"\left\{ \begin{array}{ll} "
    + r" \\ ".join(case_latexes)
    + r" \end{array} \right."
)
        
self._subject_latexes.pop()
return latex
```

and you would also have

```
    def visit_MatchValue(self, node: ast.MatchValue) -> str:
        """Visit a MatchValue node."""
        latex = self._expression_codegen.visit(node.value)
        return self._subject_latexes[-1] + " = " + latex
```

A stack is only required if nested match statements is supported, which it seems like it is, although I have no idea if it's valid $\LaTeX$ output? In any case this situation I think necessitates a test for nested match statements (assuming they're supported), in order to test that in `visit_MatchValue`, the LHS takes the appropriate `_subject_latex`, i.e. the inner one should take the inner `subject` while the outer one takes the outer.

Reply:
I think the @ZibingZhang 's solution is desirable at this point.

One minor point is that the private member `_subject_latexes` sounds weird for every other visitors, more specific name (like `_match_subject_stack`) would be better.